packages/devtools_app/lib/src/shared/charts/treemap.dart

```diff
@@ -464,71 +474,162 @@ class _TreemapState extends State<Treemap> {
     );
   }
 
-  Column buildTreemapFromRoot(BuildContext context) {
+  Column buildTreemapFromRoot() {
+    final rootNode = widget.rootNode!;
+    final child = _TreeMapCell(
+      treeMapHeight: widget.height,
+      node: rootNode,
+    );
     return Column(
       children: [
-        if (widget.isOutermostLevel) buildTitleText(context),
+        if (widget.isOutermostLevel)
+          _TitleBar(
+            rootNode: widget.rootNode!,
+            isOutermostLevel: widget.isOutermostLevel,
+            onRootChangedCallback: widget.onRootChangedCallback,
+          ),
         Expanded(
           child: widget.isOutermostLevel
-              ? buildCell()
-              : buildSelectable(child: buildCell()),
+              ? child
+              : _SelectableTreemapNode(
+                  node: rootNode,
+                  onRootChangedCallback: widget.onRootChangedCallback,
+                  child: child,
+                ),
         ),
       ],
     );
   }
 
-  Container buildCell() {
+  /// Checks if the touch point of the given [details] is overlapping with
+  /// a cell in [positionedCells].
+  ///
+  /// If so, saves the matching hoveredNode.
+  void _onHover(
+    PointerHoverEvent event,
+    List<PositionedCell> positionedCells,
+  ) {
+    final x = event.localPosition.dx;
+    final y = event.localPosition.dy;
+    final touchPoint = Offset(x, y);
+    // TODO(peterdjlee): Optimize with more efficient algorithm to find the overlapping cell.
+    //                   Currently O(positionedCells.length) but an optimized algorithm with
+    //                   O(log(positionedCells.length)) is possible.
+    for (final cell in positionedCells) {
+      if (cell.rect.contains(touchPoint)) {
```

Comment:
Nit: should we break this loop if we find a match?

Reply:
There were no logic changes in this CL, so if there's a logic bug here we should address separately.